sys/auto_init/usb/auto_init_usb.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2018 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup     sys_auto_init
+ * @{
+ * @file
+ * @brief       initializes USBUS, usb devices and handlers
+ *
+ * This auto initialization for USBUS is designed to cover the common use case
+ * of a single usb peripheral. An USBUS instance is started with USB function
+ * handlers based on which module is compiled in.
+ *
+ * If this doesn't suit your use case, a different intialization function can
+ * to be created based on this initialization sequence.
+ *
+ * @author  Koen Zandberg <koen@bergzand.net>
+ * @}
+ */
+
+#include "usb/usbus.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+
+static char _stack[USBUS_STACKSIZE];
+static usbus_t usbus;
+
+void auto_init_usb(void)
+{
+    /* Get driver context */
+    usbdev_t *usbdev = usbdev_get_ctx(0);
```

Comment:
Isn't better to retrieve the USB config from a `periph_conf.h` file and put a loop  `for(int i=0; i<USB_NUMOF;i++)` ?
Otherwise the current state will force us the support only one USB interface at a time.

Reply:
Can I claim that this is intentional at the moment:

> The auto_init as contributed is designed as a simple solution for the common case. It is intentionally not designed to solve all use cases.

On a serious note, as soon as a device has multiple USB phy's, the whole auto_init becomes a lot more complex as it would have to instantiate device functions per USB phy. This could or could not be what the developer intends. Furthermore, one phy might be high speed and the other might be full speed (I think some stm32f4 devices have this). This complicates the situation as an ethernet (CDC-ECM) should probably be on the high speed and the theoretical HID can be on the full speed.
TL,DR: automatically initializing multiple USB peripherals is a bit too complex for a simple auto_init module IMHO

Reply:
Or just loop the usbdev_get_ctx(i) and check it against USB_NUMOF defined in a periph_conf.h

Reply:
But maybe I should put my whole reasoning above in a document if it isn't somewhere already

Reply:
It is already state above in the same file. Indeed, this can be improve later when we support lowlevel driver for STM32F4 or other MCUs with at least 2 phy